### PR TITLE
Alerting: Set mimir implementation in jsonData by default when creating a new a…

### LINKED ISSUE #80204: Alerting: Can't send alerts to Grafana Cloud AM from GMA using Mimir implementation datasource
Sending alerts from Grafana Managed Alerts (GMA) to Grafana Cloud Alertmanager (GCA) appears to be broken when the GCA is configured as a datasource using the Mimir implementation.

In the following screenshots, we can see that despite the health check, the request returns 404 Not Found:

image
image

However, if we change implementation from Mimir to Prometheus, and then add `/alertmanager` to the URL, then it works:

image
image

Adding `/alertmanager` to the URL without changing the implementation from Mimir to Prometheus however causes the health check to fail:

image

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

In the UI, when creating a new alert manager data source, we decided to default to `mimir` implementation. But we were not setting the implementation field in the `jsonData` we send to the BE when saving the data source.
This leads to have a wrong alert routing (actually not working at all) , because in the BE side , the mimir implementation is not detected (as it's empty).

This is what happens:
1. User clicks new data source /alert manager type.
2. this creates immediately a data source with an `empty jsonData` field
3. UI decides that as the jsonData.implementation is empty, we default to `mimir` implementation. But it does not update the field in the UI, we only update this field when selecting an option in the implementation dropdown. 
4. BE receives an empty implementation, and it does not default to Mimir :x: => inconsistency

This PR updates the default implementation to `mimir` field in the jsonData if the implementation is not touched (mimir by default)

Fixes https://github.com/grafana/grafana/issues/80204

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/alertmanager/ConfigEditor.tsx [hunk 1]
```diff
@@ -33,6 +33,19 @@ const IMPL_OPTIONS: Array<SelectableValue<AlertManagerImplementation>> = [
...
+  // As we default to Mimir, we need to make sure the implementation is set from the start
+  useEffect(() => {
+    if (!options.jsonData.implementation) {
+      onOptionsChange({
+        ...options,
+        jsonData: {
+          ...options.jsonData,
+          implementation: AlertManagerImplementation.mimir,
+        },
+      });
```
Comment (tomratcliffe) [inline public/app/plugins/datasource/alertmanager/ConfigEditor.tsx:45]: Are we able to use `produce` to make the change to the object instead?

Comment (soniaAguilarPeiron) [reply public/app/plugins/datasource/alertmanager/ConfigEditor.tsx:45]: sure!

Comment (soniaAguilarPeiron) [reply public/app/plugins/datasource/alertmanager/ConfigEditor.tsx]: done!

**Comment by konrad147** [comment]
The default value for the implementation field seems to be wrong https://github.com/grafana/grafana/blob/ff390676057c370fb45501f9a6aa26287ddd313c/public/app/plugins/datasource/alertmanager/ConfigEditor.tsx#L46

It should always take the value from `jsonData.implementation`

File: public/app/plugins/datasource/alertmanager/ConfigEditor.tsx [hunk 2]
```diff
@@ -33,6 +33,17 @@ const IMPL_OPTIONS: Array<SelectableValue<AlertManagerImplementation>> = [
 export const ConfigEditor = (props: Props) => {
   const { options, onOptionsChange } = props;
 
+  // As we default to Mimir, we need to make sure the implementation is set from the start
+  useEffect(() => {
```
Comment (konrad147) [inline public/app/plugins/datasource/alertmanager/ConfigEditor.tsx:37]: I wonder if there is no better place somewhere in `public/app/plugins/datasource/alertmanager/DataSource.ts` Alertmanager data source file for this type of initialization, but I'm not very familiar with data sources

Comment (soniaAguilarPeiron) [reply public/app/plugins/datasource/alertmanager/ConfigEditor.tsx:37]: yep, I was wondering the same question and I didn't find another way to do it

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**